src/ip/Address.cc

```diff
@@ -707,7 +707,9 @@ Ip::Address::FreeAddr(struct addrinfo *&ai)
 {
     if (ai == nullptr) return;
 
-    if (ai->ai_addr) delete ai->ai_addr;
+    if (ai->ai_addr) {
+        delete (struct sockaddr_storage*)ai->ai_addr;
+    }
```

Comment:
A few issues here:
 1) C++ `delete` is null-safe. you do not need the if-statement.
 2) please use C++ casting
```suggestion
    delete reinterpret_cast<struct sockaddr_storage *>(ai->ai_addr);
```